### PR TITLE
Reduce load of files versions preview loading

### PR DESCRIPTION
**Pull request by artonge**
- [x] Provide hasPreview in files_versions DAV API to prevent request on non-existing previews
- [x] Set properties of the `` element to reduce preview loading to what the browser think is necessary
- [x] Load preview only when versions tab is active.

Especially useful when there are a lot of versions.

### CODE REVIEW

File: apps/files_versions/src/components/Version.vue [hunk 1]
```diff
@@ -23,11 +23,15 @@
...
+				<div v-if="!loadPreview && !previewLoaded" class="version__image" />
+				<img v-else-if="isCurrent || version.hasPreview"
 					:src="previewURL"
 					alt=""
+					decoding="async"
+					fetchpriority="low"
+					loading="lazy"
 					class="version__image"
-					@error="previewError = true">
+					@load="previewLoaded = true">
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/files_versions/src/components/Version.vue:34]: 1. It is provided by `VersionTab.vue`
2. `previewLoaded` makes sense no?
3. If one of them is `true` it is rendered. I can rewrite to `loadPreview || previewLoaded` if you think that it is more readable.
4. See #3

**Review by skjnldsv — APPROVED** [review]

File: apps/files_versions/src/components/Version.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_versions/src/components/Version.vue:34]: > I am pushing the rewrite of 3, it's much more readable sweat_smile

ngl, I got confused for a sec :see_no_evil:

File: apps/files_versions/lib/Sabre/Plugin.php [hunk 2]
```diff
@@ -89,8 +94,25 @@ public function afterGet(RequestInterface $request, ResponseInterface $response)
...
 	public function propFind(PropFind $propFind, INode $node): void {
+		$user = $this->userSession->getUser();
+
 		if ($node instanceof VersionFile) {
 			$propFind->handle(self::VERSION_LABEL, fn() => $node->getLabel());
+
+			if ($user !== null) {
+				$propFind->handle(FilesPlugin::HAS_PREVIEW_PROPERTYNAME, function () use ($node, $user) {
+					try {
+						$this->versionManager->getVersionFile(
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/files_versions/lib/Sabre/Plugin.php]: Go idea, I changed the logic to use `$previewManager->isMimeSupported(...)`

**Review by Pytal — APPROVED** [review]

**Pull request merged**